spec/classes/init_spec.rb

```diff
@@ -260,6 +260,52 @@
 
   end
 
+  context 'with selboolean specified as a hash' do
+    let(:facts) { { :osfamily => 'RedHat' } }
+    let :params do
+      {
+        :selboolean_hiera_merge => 'false',
+        :selboolean => {
+          'nfs_export_all_ro' => {
+            'value' => 'on',
+          },
+          'nfs_export_all_rw' => {
+            'persistent' => true,
+            'value'      => 'on',
+          },
+        }
+      }
+    end
+
+    it { should contain_class('types') }
+    it {
+      should contain_selboolean('nfs_export_all_ro').with({
+        'value' => 'on',
+      })
+    }
+    it {
+      should contain_package('nfs_export_all_rw').with({
+        'persistent' => 'true',
+        'value' => 'on',
+      })
+    }
+  end
+
+  context 'with selboolean specified incorrectly' do
```

Comment:
specified as an invalid type - saying it is specified incorrectly does not tell us how it is incorrectly specified. It could be the correct type but other validation is being used.